test/python/transpiler/test_optimize_phase_shift_gates.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Tests for pass marging/canceling phase-shift gates and cnot cancellation."""
+
+import numpy as np
+import qiskit
+from qiskit import QuantumCircuit, QuantumRegister
+import copy
+from qiskit import BasicAer, execute
+from qiskit.quantum_info import Pauli, state_fidelity, basis_state, process_fidelity
+
+
+class TestOptimizePhaseShiftGates():
+    """Test the TestOptimizePhaseShiftGates pass."""
+
+    def test_pass_optimize_phase_shift_gates():
+        """Test the cx TestOptimizePhaseShiftGates pass.
+
+        It should merge/cancel phase-shift gates and perform cnot cancellation.
+        """
+        n = 6
+        q = QuantumRegister(n, 'q')
+        circ = QuantumCircuit(q)
+        for i in range(150):
+            x = np.random.randint(9)
+            if x == 0:
+                circ.t(q[np.random.randint(n)])
+            elif x == 1:
+                circ.h(q[np.random.randint(n)])
+            elif x == 2:
+                circ.tdg(q[np.random.randint(n)])
+            elif x == 3:
+                circ.s(q[np.random.randint(n)])
+            elif x == 4:
+                circ.sdg(q[np.random.randint(n)])
+            elif x == 5:
+                circ.z(q[np.random.randint(n)])
+            elif x == 6:
+                circ.u1(np.random.random(2)[0], q[np.random.randint(n)])
+            elif x == 7:
+                y = np.random.randint(n-1)
+                circ.cx(q[y], q[y+1])
+            elif x == 8:
+                y = np.random.randint(n-1)
+                circ.cx(q[y+1], q[y])
+        circ_dag = qiskit.converters.circuit_to_dag(circ)
+        circ_dag = qiskit.transpiler.passes.OptimizePhaseShiftGates.run(circ_dag,circ_dag)
+        circ_new = qiskit.converters.dag_to_circuit(circ_dag)
+        backend_sim = BasicAer.get_backend('unitary_simulator')
+        result_original = execute(circ, backend_sim).result()
+        unitary_original = result_original.get_unitary(circ)
+        backend_sim = BasicAer.get_backend('unitary_simulator')
+        result_optimized = execute(circ_new, backend_sim).result()
+        unitary_optimized = result_optimized.get_unitary(circ_new)
+        if np.sum(np.abs(unitary_optimized-unitary_original)) > 0.001:
```

Comment:
Could you update your testing to use assert statements? It should just be a case of updating these end few lines, for example `assertEqual(unitary_original, unitary_optimized)`. There is a good example of how to do it [here](https://github.com/Qiskit/qiskit-terra/blob/master/test/python/transpiler/test_collect_2q_blocks.py)

Reply:
Thank you for the comment. We implemented everything as requested. Hopefully it is ok now.